File: Create/Prototype/Prototype.java
// 1. Prototype 接口
public interface Graphic extends Cloneable {
    void draw(int[] pos);
    Graphic clone();
}

// 2. 具体原型
public class WholeNote implements Graphic {
    public void draw(int[] pos) {
        System.out.println("Draw WholeNote at (" + pos[0] + "," + pos[1] + ")");
    }
    public WholeNote clone() {
        try { return (WholeNote) super.clone(); }
        catch (CloneNotSupportedException e) { throw new RuntimeException(e); }
    }
}

public class HalfNote implements Graphic {
    public void draw(int[] pos) {
        System.out.println("Draw HalfNote at (" + pos[0] + "," + pos[1] + ")");
    }
    public HalfNote clone() {
        try { return (HalfNote) super.clone(); }
        catch (CloneNotSupportedException e) { throw new RuntimeException(e); }
    }
}

// 3. Client（工具）
public class GraphicTool {
    private final Graphic prototype;

    public GraphicTool(Graphic prototype) {
        this.prototype = prototype;
    }

    public void clickAt(int[] pos) {
        Graphic newGraphic = prototype.clone(); // 关键：克隆
        newGraphic.draw(pos);
    }

    public static void main(String[] args) {
        GraphicTool wholeTool = new GraphicTool(new WholeNote());
        wholeTool.clickAt(new int[]{120, 200});
    }
}

File: Create/Prototype/Prototype.py
import copy
from abc import ABC, abstractmethod

# 1. Prototype 接口
class Graphic(ABC):
    @abstractmethod
    def draw(self, pos): ...
    @abstractmethod
    def clone(self) -> "Graphic": ...

# 2. 具体原型
class WholeNote(Graphic):
    def draw(self, pos):
        print(f"Draw WholeNote at {pos}")
    def clone(self):
        return copy.deepcopy(self)

class HalfNote(Graphic):
    def draw(self, pos):
        print(f"Draw HalfNote at {pos}")
    def clone(self):
        return copy.deepcopy(self)

# 3. Client（工具）
class GraphicTool:
    def __init__(self, prototype: Graphic):
        self._prototype = prototype

    def click_at(self, pos):
        new_graphic = self._prototype.clone()  # 关键：克隆
        new_graphic.draw(pos)

if __name__ == "__main__":
    tool = GraphicTool(WholeNote())
    tool.click_at((120, 200))